tests/test_tokenization_fast.py

```diff
@@ -93,6 +95,51 @@ def test_training_new_tokenizer_with_special_tokens_change(self):
                     self.tmpdirname = tmpdirname_orig
 
 
+@require_tokenizers
+class TokenizerVersioningTest(unittest.TestCase):
+    def test_local_versioning(self):
+        tokenizer = AutoTokenizer.from_pretrained("bert-base-cased")
+        json_tokenizer = json.loads(tokenizer._tokenizer.to_str())
+        json_tokenizer["model"]["vocab"]["huggingface"] = len(tokenizer)
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tokenizer.save_pretrained(tmp_dir)
+            json.dump(json_tokenizer, open(os.path.join(tmp_dir, "tokenizer.4.0.0.json"), "w"))
+
+            # This should pick the new tokenizer file as the version of Transformers is > 4.0.0
+            new_tokenizer = AutoTokenizer.from_pretrained(tmp_dir)
+            self.assertEqual(len(new_tokenizer), len(tokenizer) + 1)
+            json_tokenizer = json.loads(new_tokenizer._tokenizer.to_str())
+            self.assertIn("huggingface", json_tokenizer["model"]["vocab"])
+
+            # Will need to be adjusted if we reach v42 and this test is still here.
+            # Should pick the old tokenizer file as the version of Transformers is < 4.0.0
```

Comment:
Let's think about it next century :)

Reply:
when we near AGI this library will publish itself and who knows which version number it will pick